recipes/rmlui/3.3/conanfile.py

```diff
@@ -0,0 +1,100 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class RmluiConan(ConanFile):
+    name = "rmlui"
+    description = "RmlUi - The HTML/CSS User Interface Library Evolved"
+    homepage = "https://github.com/mikke89/RmlUi"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "https://github.com/mikke89/RmlUi/blob/master/LICENSE"
+    topics = ("conan", "css", "gui", "html", "lua", "rmlui")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "build_lua_bindings": [True, False],
+        "build_samples": [True, False],
+        "disable_rtti_and_exceptions": [True, False],
+        "enable_precompiled_headers": [True, False],
+        "enable_tracy_profiling": [True, False],
+        "fPIC": [True, False],
+        "no_font_interface_default": [True, False],
+        "no_thirdparty_containers": [True, False],
```

Comment:
Typically we try to avoid 'vendored' deps that are embedded into a project because it prevents consumers from choosing the version themselves and benefiting from updates

https://github.com/mikke89/RmlUi/blob/6e0e3011b23f617f1f301cee0dd595c7ae20e650/Include/RmlUi/Core/Containers/LICENSE.txt

Have you looked where these external projects come from? is it possible to offer them as `requirements` slash make them available from CCI?

Reply:
I do sympathize with the sentiment that you're trying to convey here and I did some research on the matter.

Basically, there are three separate non-STL containers that are put to use if the "RMLUI_NO_THIRDPARTY_CONTAINERS" compile definition is not set, otherwise ordinary STL containers come into play. This means that introducing dependencies via conan would definitely require some patching. However, given the way all of this is implemented in the library (Include/RmlUi/Core/Types.h:40) such a thing would be very easy to do.

As for the containers themselves:
|Name|Version embedded|Versions available on conan|
|-|-|-|
|chobo-flat-map|v1.01|none|
|chobo-flat-set|v1.00|none|
|robin-hood-hashing|3.5.0|>=3.7.0|

Since in each of the cases there either isn't an available package or the embedded version is too old, I'm of the opinion that it's better left untouched for now. Still, I'm already thinking about making a recipe for RmlUi's upcoming version, which has robin-hood-hashing 3.9.0 embedded judging from the master branch. In that case, the change could be made without the risk of breaking something.

In light of the above, would such a futuristic outlook be acceptable?

Reply:
Thank you for the detailed response!

Given none of the deps are available and individual patching... I think its acceptable in this case.. there's an option to avoid them which is suitable